lib/kennel.rb

```diff
@@ -125,6 +135,20 @@ def api
       @api ||= Api.new
     end
 
+    def p_parts
+      @p_parts ||= Rosarium::Promise.execute { parts(plain: true) }
+    end
+
+    def p_definitions
+      @p_definitions ||= Rosarium::Promise.execute { definitions(plain: true) }
+    end
```

Comment:
idk how this is supposed to be faster, old code does them in serial ... and we can't really do them in parallel right ?
```
      parts = generated
      PartsSerializer.new(filter: filter).write(parts) if ENV["STORE"] != "false" # quicker when debugging
      parts
```

Reply:
The faster part (only by a second or two) is that the `PartsSerializer` doesn't need to wait for the `definitions` to be ready. So writing to `generated` can happen in parallel with downloading from the API.

Reply:
ah PartsSerializer did not run in the thread ...

but so all we need is:
```
todo = []
todo.push :parts, :definitions if needs_syncing
todo.push :serialize_parts if generate?
Utils.parallel(todo) { |m| send m, plain: true }
```

Reply:
corrected the code

Reply:
That would be a minimal way of achieving the desired change. I think it might be a bit short-sighted though – but without me pointing to _further_ benefits in this direction, "short-sighted" may be just fine :-)